arch/arm/src/cmake/armclang.cmake

```diff
@@ -116,8 +116,8 @@ if(CONFIG_STACK_USAGE_WARNING AND NOT "${CONFIG_STACK_USAGE_WARNING}" STREQUAL
   add_compile_options(-Wstack-usage=${CONFIG_STACK_USAGE_WARNING})
 endif()
 
-if(CONFIG_SCHED_GCOV)
-  add_compile_options(-fprofile-generate -ftest-coverage)
+if(CONFIG_COVERAGE_ALL)
+  add_compile_options(-fprofile-instr-generate -fcoverage-mapping)
```

Comment:
do we need different option for full and mini

Reply:
I think we don't need to do this for now, we can keep the same options as mini.profile. Other options are slightly different, we can explore them when we need them